src/clipping/weileratherton.jl

```diff
@@ -0,0 +1,203 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    WeilerAtherton()
+
+The Weiler-Atherton algorithm for clipping polygons.
+
+## References
+
+* Weiler, Kevin and Atherton, Peter. 1977. [Hidden surface removal using polygon area sorting]
+  (https://dl.acm.org/doi/10.1145/563858.563896)
+"""
+struct WeilerAtherton <: ClippingMethod end
+
+function clip(poly::Polygon{Dim,T}, other::Geometry{Dim,T}, method::WeilerAtherton) where {Dim, T}
+  if hasholes(poly)
+    outers = clip(rings(poly)[1], boundary(other), method)
+    inners = vcat([clip(r, boundary(other), method) for r in rings(poly)[2:end]]...)
+    if isempty(outers)
+      return nothing
+    end
+
+    g = PolyArea{Dim,T}[]
+    # build a PolyArea to each outer ring generated
+    for o in outers
+      ringsₒ = [o]
+      # find inner rings inside the outer ring o 
+      for i in inners
+        if any([sideof(v, o) == IN for v in vertices(i)])
+          push!(ringsₒ, i)
+        end
+      end
+      push!(g, PolyArea(ringsₒ))
+    end
+    Multi(g)
+  else
+    outers = clip(rings(poly)[1], boundary(other), method)
+    if isempty(outers)
+      nothing
```

Comment:
I have the impression that this code could be much cleaner. @eliascarv can you take a look later and help with the cleanup?